include/onnxruntime/core/framework/op_kernel_info.h

```diff
@@ -40,6 +40,8 @@ class OpKernelInfo : public OpNodeProtoHelper<ProtoHelperNodeContext> {
 
   bool TryGetConstantInput(int input_index, const Tensor** constant_input_value) const;
 
+  bool TryGetConstantInput(const std::string& name, const Tensor** constant_input_value) const;
```

Comment:
>const std::string& name, [](start = 27, length = 24)

why you want to compare with name? we should already have the name to idx mapping in sesseion_state, all the apis in OpKernel is idx based, as compare string will hurt perf.

Reply:
This is only called in kernel constructor. The reason to use name is that there are initializers in subgraph that are not listed as the subgraphs' inputs, and using name will make sure they could be found.

---
In reply to: [294412651](https://github.com/microsoft/onnxruntime/pull/888#discussion_r294412651) [](ancestors = 294412651)